src/udev/udev-builtin-path_id.c

```diff
@@ -635,8 +636,24 @@ static int find_real_nvme_parent(sd_device *dev, sd_device **ret) {
                 return r;
 
         /* If the 'real parent' is (still) virtual, e.g. for nvmf disks, refuse to set ID_PATH. */
-        if (path_startswith(devpath, "/devices/virtual/"))
-                return -ENXIO;
+        if (path_startswith(devpath, "/devices/virtual/")) {
+		const char *syspath;
+		const char *addr, *fc_pn, *name, *nvme_pn;
+		struct sd_device *fcdev = NULL, *fc_parent = NULL;
+
+		sd_device_get_sysattr_value(nvme, "address", &addr);
+		nvme_pn = strrchr(addr, '-') + 1;
```

Comment:
Missing error handling:
```c
nvme_pn = strrchr(addr, '-');
if (!nvme_pn)
        return -ENXIO;

nvme_pn++;
```

Reply:
Also, please make the parser more strict?
First, extract the host interface field (`"host_iface="`) from `address`, then cut the unnecessary part.
`extract_first_word()` can be used.

Reply:
Could you provide an example of `address` attribute?

Reply:
localhost:/sys/class/nvme-fabrics/ctl/nvme1 # cat address
traddr=nn-0x50050768100035e5:pn-0x50050768101935e5,host_traddr=nn-0x200000109bec9b93:pn-0x100000109bec9b93